ci/benchmark/build.sh

```diff
@@ -107,17 +112,59 @@ if (( ${ERRORCODE} != 0 )); then
     exit ${ERRORCODE}
 fi
 
+
+# Concatenate dependency arrays, convert to JSON array,
+# and remove duplicates.
+X=("${CUGRAPH_DEPS[@]}" "${LIBCUGRAPH_DEPS[@]}")
+DEPS=$(printf '%s\n' "${X[@]}" | jq -R . | jq -s 'unique')
+
+# Build object with k/v pairs of "dependency:version"
+DEP_VER_DICT=$(jq -n '{}')
+for DEP in $(echo "${DEPS}" | jq -r '.[]'); do
+  VER=$(conda list | grep "^${DEP}" | awk '{print $2"-"$3}')
+  DEP_VER_DICT=$(echo "${DEP_VER_DICT}" | jq -c --arg DEP "${DEP}" --arg VER "${VER}" '. + { ($DEP): $VER }')
+done
+
+# Pass in an array of dependencies to get a dict of "dependency:version"
+function getReqs() {
+  local DEPS_ARR=("$@")
+  local REQS="{}"
+  for DEP in "${DEPS_ARR[@]}"; do
+    VER=$(echo "${DEP_VER_DICT}" | jq -r --arg DEP "${DEP}" '.[$DEP]')
+    REQS=$(echo "${REQS}" | jq -c --arg DEP "${DEP}" --arg VER "${VER}" '. + { ($DEP): $VER }')
+  done
+
+  echo "${REQS}"
+}
+
+REQS=$(getReqs "${CUGRAPH_DEPS[@]}")
+
+BECHMARK_META=$(jq -n \
+  --arg NODE "${NODE_NAME}" \
+  --arg MINOR_VER "${MINOR_VERSION}" \
+  --argjson REQS "${REQS}" '
+  {
+    "machineName": $NODE,
+    "commitBranch": $MINOR_VER,
+    "requirements": $REQS
+  }
+')
+
 logger "Running Benchmarks..."
 cd $BENCHMARKS_DIR
 set +e
 time pytest -v -m "small and managedmem_on and poolallocator_on" \
     --benchmark-gpu-device=0 \
     --benchmark-gpu-max-rounds=3 \
-    --benchmark-asv-metadata="machineName=${NODE_NAME}, commitBranch=branch-${MINOR_VERSION}" \
-    --benchmark-asv-output-dir=${ASVRESULTS_DIR}
+    --benchmark-asv-output-dir="${ASVRESULTS_DIR}" \
+    --benchmark-asv-metadata="${BECHMARK_META}"
+
+
 
 EXITCODE=$?
 
+# libcugraph reqs
+# REQS=$(getReqs "${LIBCUGRAPH_DEPS[@]}")
```

Comment:
Is this comment needed for some reason or can it be removed?

Reply:
oh, I think I see, I guess this is also a placeholder?  I usually like that but cugraph will (probably) never have C++ benches.  If you choose to leave it, Can you add a comment to the future maintainer that explains it should be uncommented and used when libcugraph benchmarks are added?

Reply:
i'll add a comment about this